src/frequenz/sdk/actor/power_distributing/battery_pool.py

```diff
@@ -0,0 +1,125 @@
+            max_data_age_sec: If component stopped sending data, then
+                this is the maximum time when its last message should be considered as
+                valid. After that time, component won't be used until it starts sending
+                data.
+            max_blocking_duration_sec: This value tell what should be the maximum
+                timeout used for blocking failing component.
+        """
+        self._batteries = {
+            id: BatteryUtils(id, max_data_age_sec, max_blocking_duration_sec)
+            for id in battery_ids
+        }
+        self._run_method_called: bool = False
+
+    async def init(self) -> None:
+        """Run battery pool."""
+        tasks = [
+            asyncio.create_task(battery.init()) for battery in self._batteries.values()
+        ]
+        await asyncio.wait(tasks, return_when=ALL_COMPLETED)
+        self._run_method_called = True
+
+    def get_working_batteries(self, battery_ids: Set[int]) -> Set[int]:
+        """Get subset of battery_ids with working batteries.
+
+        Args:
+            battery_ids: batteries ids
+
+        Raises:
+            RuntimeError: If `run` method was not called at the beginning to
+                initialized object.
+            KeyError: If any battery in the given batteries is not in the pool.
+
+        Returns:
+            Subset of given batteries with working batteries.
+        """
+        if not self._run_method_called:
+            raise RuntimeError(
+                "`run` method not called or not awaited. Run it before first use"
+            )
+
+        working: Set[int] = set()
+        uncertain: Set[int] = set()
+        for bat_id in battery_ids:
+            if bat_id not in battery_ids:
+                ids = str(self._batteries.keys())
+                raise KeyError(f"No battery {bat_id} in pool. All batteries: {ids}")
+            battery_status = self._batteries[bat_id].get_state()
+            if battery_status == WorkingState.WORKING:
+                working.add(bat_id)
+            elif battery_status == WorkingState.UNCERTAIN:
+                uncertain.add(bat_id)
+
+        if len(working) > 0:
+            return working
+
+        _logger.warning(
+            "There are no working batteries in %s. Falling back to using uncertain batteries %s.",
+            str(battery_ids),
+            str(uncertain),
+        )
+        return uncertain
+
+    def update_last_request_status(self, result: Result):
+        """Update batteries in pool based on the last result from the request.
+
+        Args:
+            result: Summary of what batteries failed and succeed in last request.
+
+        Raises:
+            RuntimeError: If `run` method was not called at the beginning to initialize
+                object.
+        """
+        if not self._run_method_called:
+            raise RuntimeError(
+                "`run` method not called or not awaited. Run it before first use"
```

Comment:
`_init_called`?  and `init method not called ...`.  In the docstring also.

Reply:
@leandro-lucarella-frequenz 
> Making instantiation and async init one step will also avoid having to do all these checks 

https://github.com/frequenz-floss/frequenz-sdk-python/pull/117#discussion_r1050787860

Reply:
> _init_called? and init method not called .... In the docstring also.

Done